### PR TITLE
compilationConstants Implementation

### LINKED ISSUE #15: Implement compilation constants in Makejs
(See CompilationConstants.ts for details)

### PR DESCRIPTION
**Pull request by kaaninel**
Fixes #15

### CODE REVIEW

**Comment by paul-go** [comment]
- You can use String.repeat instead of a regex
- I don't think this needs to be async ... it's a build tool, I don't think non-blocking IO matters.
- There's some code formatting problems here, but I think you probably don't have ESLint setup properly.
- I think you've got whitespace trimming on your editor (looks like the .editorconfig isn't working)

File: Makejs/sample/a.ts [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+
+if ("DEBUG")
+{
+	console.log("(chuckles) I'm in danger.");
+}
+
+if ("MODERN") 
+{
+	console.log("I'm here!");
+}
```
Comment (paul-go) [inline Makejs/sample/a.ts]: ESLint is supposed to complain when a file doesn't end in a blank line ... not sure what's going on here

File: Makejs/sample/make.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-make.on(async () =>
+make.on(async () => 
 {
 	await make.typescript("./tsconfig.json", {
 		compilerOptions: {
 			outDir: "./build"
 		}
 	});
-	
+	make.compilationConstants("./build/a.js", {
```
Comment (paul-go) [inline Makejs/sample/make.js]: I'd probably put a blank line ontop of this ... it's a new "paragraph"

File: Makejs/source/CompilationConstants.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
 	 */
-	export function compilationConstants(
-		file: string,
-		constants: { [key: string]: boolean; })
-	{
-		// TODO: Provide the implementation.
+	export function compilationConstants( 
+		file: string, 
+		constants: { [key: string]: boolean }
+	) {
```
Comment (paul-go) [inline Makejs/source/CompilationConstants.ts]: The ) is supposed to go on the previous line here

Comment (kaaninel) [reply Makejs/source/CompilationConstants.ts]: my eslint nags about everything except actualy problems :)

**Pull request merged**